syft/workers/websocket_client.py

```diff
@@ -113,12 +120,41 @@ def list_objects_remote(self):
     def objects_count_remote(self):
         return self._send_msg_and_deserialize("objects_count")
 
-    def fit(self, dataset_key, **kwargs):
+    async def fit(self, dataset_key, **kwargs):
```

Comment:
this function is not easy to follow, maybe break this into multiple functions. Also, I think `synchronous_fit` should be called `fit` and this one called `async_fit`. Or even create a wrapper function `fit` with a parameter `async` with default value = False.

Reply:
A wrapper function is not possible, as the whole chain of async functions needs the async keyword. 
But I can do the renaming.